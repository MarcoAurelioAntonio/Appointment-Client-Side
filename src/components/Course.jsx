import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Course = ({ data }) => (
  <Link className="course-item" to="/add-reservation" state={data}>
    <p className="course-title">{ data.name }</p>
    <p className="course-label">{ data.course_type }</p>
    <p className="course-price">{ data.price }</p>
    <div className="course-duration">
      <p className="course-label">
        From:
        { data.start_date }
      </p>
      <p className="course-label">
        To:
        { data.end_date }
      </p>
    </div>
  </Link>
);

Course.propTypes = {
  data: PropTypes.instanceOf(Object).isRequired,
};

export default Course;
