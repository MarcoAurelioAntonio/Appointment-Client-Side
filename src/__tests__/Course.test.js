import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Course from '../components/Course';

describe('Tests for <Course /> component', () => {
  const courseItem = {
    id: 9,
    name: 'Nelson asdadsasdasdads',
    description: 'dsadasdasdasdasdasdadsasdasdasdasddas',
    start_date: '2023-06-21',
    end_date: '2023-06-24',
    price: '20.0',
    course_type: 'asdads',
    image_url: '',
  };

  it('In the Course screen should be displayed the name of the course', () => {
    render(
      <BrowserRouter>
        <Course data={courseItem} />
        ,
      </BrowserRouter>,
    );
    const element = screen.getByText(courseItem.name);
    expect(element.textContent).toBe(courseItem.name);
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(
      <BrowserRouter><Course data={courseItem} /></BrowserRouter>,
    ).toJSON()).toMatchSnapshot();
  });
});
