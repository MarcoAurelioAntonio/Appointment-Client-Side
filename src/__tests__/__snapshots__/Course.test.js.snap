// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Course /> component Should match the snapshot 1`] = `
<a
  className="course-item"
  href="/courses/9"
  onClick={[Function]}
>
  <div
    className="image-container"
  >
    <img
      alt="Course-img"
      className="course-image-nelson"
      src="http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBDZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--b4fee963a3f0b6b5be716592d385e19264fa88ca/Screenshot%202023-06-20%20at%2012.44.39%20PM.png"
    />
  </div>
  <p
    className="course-title"
    data-testid="name-9"
  >
    Nelson asdadsasdasdads
  </p>
  <div
    className="separator"
  >
    . . . . . . . . . . .
  </div>
  <p
    className="course-label"
  >
    dsadasdasdasdasdasdadsasdasdasdasddas
  </p>
  <p
    className="course-price"
  >
    Course-Type: asdads
  </p>
  <p
    className="course-price"
  >
    Price: 20.0 $
  </p>
  <div
    className="course-duration"
  >
    <p
      className="course-price"
    >
      Duration: 0 hours
    </p>
  </div>
</a>
`;
